src/main/java/DukeException.java

```diff
@@ -0,0 +1,11 @@
+/**
+ * An exception that is thrown when Duke encounters an error.
+ */
+public class DukeException extends Exception {

```

Comment:
Good use of OOP through defining subclasses of DukeException to handle different types of error.